agent/app/agent_capability_unix.go

```diff
@@ -112,5 +121,35 @@ func (agent *ecsAgent) appendAppMeshCapabilities(capabilities []*ecs.Attribute)
 }
 
 func (agent *ecsAgent) appendTaskEIACapabilities(capabilities []*ecs.Attribute) []*ecs.Attribute {
-	return appendNameOnlyAttribute(capabilities, attributePrefix+taskEIAAttributeSuffix)
+
+	capabilities = appendNameOnlyAttribute(capabilities, attributePrefix+taskEIAAttributeSuffix)
+
+	eiaRequiredFlags := []string{AVX, AVX2, SSE41, SSE42}
+	cpuInfo, err := utils.ReadCPUInfo(CpuInfoPath, agent.ioutil)
+	if err != nil {
+		seelog.Warnf("Unable to read cpuinfo: %v", err)
+		return capabilities
+	}
+
+	// Merge all processors' flags
```

Comment:
nit: should the utility package return a merged set of flags? Just a thought.

Reply:
I will add another helper function to do this.